### PR TITLE
For a better integration.

### PR DESCRIPTION
**Pull request by blackrez**
Now the html/css is bootsrap ready.

### CODE REVIEW

File: assets/embauche.html [hunk 1]
```diff
@@ -53,9 +53,9 @@
 				</select>.
 			</div>
 
-			<div>
-				Mon <span class="employer_type">entreprise</span>
-				<select name="jeune_entreprise_innovante">
+			<div class="form-inline">
+				Mon <span class="form-control" class="employer_type">entreprise</span>
```
Comment (MattiSG) [inline assets/embauche.html]: Illegal duplicate `class` attributes.

File: css/main.css [hunk 2]
```diff
@@ -1,4 +1,4 @@
-body {
+.embauchebody {
```
Comment (MattiSG) [inline css/main.css]: Use [BEM](https://en.bem.info).

**Comment by MattiSG** [comment]
Thanks @blackrez!  :)

I am not 100% convinced about hardcoding Bootstrap classes in this widget, though, as not everyone is using Bootstrap and that might conflict with other styles.

Taking a look at your changes, have you tried something like that, appended right after the widget include `<script>`:

```html
<script>
document.querySelector('.embauchebody form').className += 'form-inline';

Array.prototype.forEach.call(document.querySelectorAll('.embauchebody input, .embauchebody select'), function(el) {
    el.className += ' form-control';
});
</script>
```

?

Let me know how it goes, if that's of use to you I can start distributing a Bootstrap-augmented version too  :)

**Comment by blackrez** [comment]
--
  Nabil Servais
  nabil@eml.cc
 
 
 
Le Lun 27 juil 2015, à 06:29, Matti Schneider a écrit :
> Thanks @blackrez[1]!  :)


> I am not 100% convinced about hardcoding Bootstrap classes in this widget, though, as not everyone is using Bootstrap and that might conflict with other styles.


Yes, you're right, I wasn't sure about the best way to integrate bootstrap (or other css framework).
 
> Taking a look at your changes, have you tried something like that, appended right after the widget include <script>:


> <script> document.querySelector('.embauchebody form').className += 'form-inline';  Array.prototype.forEach.call(document.querySelectorAll('.embauchebody input, .embauchebody select'), function(el) {     el.className += ' form-control'; }); </script>
> ?


 
It's very effective ^^. 
 
But we need the two div https://github.com/sgmap/cout-embauche/pull/4/files#diff-f33b91bc7bc0e5da175bc361767cc1d6R4 and https://github.com/sgmap/cout-embauche/pull/4/files#diff-f33b91bc7bc0e5da175bc361767cc1d6R56 for using "form-inline" classes.
 
> Let me know how it goes, if that's of use to you I can start distributing a Bootstrap-augmented version too  :)


> 


> —
> Reply to this email directly or view it on GitHub[2]. 
> 


>  
 


Links:

  1. https://github.com/blackrez
  2. https://github.com/sgmap/cout-embauche/pull/4#issuecomment-125083927

**Comment by MattiSG** [comment]
> But we need the two div 

Then what about `Array.prototype.forEach.call(document.querySelectorAll('.embauchebody form > div', function(el) { el.className += 'form-inline'; })`?  :)

**Comment by blackrez** [comment]
This piece of code solve the main issue 

```          
Array.prototype.forEach.call(document.querySelectorAll('.embauche form > div'), function(el) { el.className += 'form-inline'; })

Array.prototype.forEach.call(document.querySelectorAll('.embauche input, .embauche select'), function(el) {     el.className += ' form-control'; });
```

How integrate this to the project?

**Comment by MattiSG** [comment]
I'll:

- [ ] Make a script of this on its own.
- [ ] Add a new webpack target.
- [ ] Document usage for Bootstrap users.

Can you please explain your usage of `.embauchebody` CSS rules though? I don't think I declared this class name, I don't see where you're introducing it, yet you're adding it in this PR… ?  :)

**Comment by blackrez** [comment]
It's typo error, I needed to remove the body rules. But I saw there is a embauche classes, it would be perfect if the rules applied to body goes to there.

**Comment by MattiSG** [comment]
Great, thanks!  :)

**Comment by blackrez** [comment]
You can check by yourself.

http://apicarto-dev.sgmap.fr/paie-carto/

Thanks a lot.

**Comment by MattiSG** [comment]
:dancer:

These URLs will go down once they are merged into `gh-pages`. I will notify you when they do.

**Comment by MattiSG** [comment]
Ok, this improvement is now the official version  :)  Thanks a lot for your beta-testing and suggestions!

Please use the [stable URLs](https://github.com/sgmap/cout-embauche#usage) now. I will erase the temporary ones from the `bootstrap` branch by tomorrow.

**Pull request closed without merging**